boa/src/property/mod.rs

```diff
@@ -34,7 +34,7 @@ pub use attribute::Attribute;
 /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty
 #[derive(Debug, Clone, Trace, Finalize)]
 pub struct DataDescriptor {
-    pub(crate) value: Value,
+    pub(crate) value: Option<Value>,
     attributes: Attribute,
 }
```

Comment:
Another way of implementing this is to have a `bool` flag `has_value` that could be accessed with a `.has_value()` method, and when create with `new_without_value` it would be `false` and value would be `undefined`. this way we don't call unwrap since calling `value()` will be `value.unwrap_or_else(Value::undefined)`. and when we need to know if it has a value we just check with the `.has_value()`.

What do you think @raskad ?

Reply:
Yes I thought about that as well. I will implement that version. Seems way cleaner.

Reply:
Done with the implementation. Somehow there is one test less passing now (`test/built-ins/Map/prototype/set/append-new-values-normalizes-zero-key.js`), but I think that one is a fluke. I don't know how that was passing with the other change in the first place. After looking at the relevant `Map.set` implementation, that one looks unrelated.